### ISSUE TITLE
RetinaNet not detecting all instance of target class

### ISSUE THREAD

**Issue by zerovirus123** [fizyr/keras-retinanet#1045]
Hello all,

I am training RetinaNet to detect tree crowns in aerial photos. I am working on a dataset with 3100 annotated images, of size 224x224. There is only class to be detected (palm tree).

![1-P17B](https://user-images.githubusercontent.com/9401015/59990487-d7872400-9632-11e9-8907-96d5ce3333db.jpg)

The classification loss drops much quicker than the regression loss, which I suppose is normal given that I only have one target class.

<img width="801" alt="training_rate" src="https://user-images.githubusercontent.com/9401015/59990297-1072c900-9632-11e9-8df6-805aede3c7f9.PNG">

However, my model missed out on a lot of target class instances during inference, as shown below.

<img width="493" alt="result1" src="https://user-images.githubusercontent.com/9401015/59990396-8414d600-9632-11e9-94ce-1d4cc59532b9.PNG">

<img width="500" alt="result2" src="https://user-images.githubusercontent.com/9401015/59990404-8b3be400-9632-11e9-89bc-9927770b709e.PNG">

My colleague suggested that there is something wrong with the classification part of the model, since there are a lot of false negatives. I think that the regression part is fine, since the trees can be localized fairly well. Another thing to note is that there are repeated bounding boxes drawn on target object. Which files should I modify to tune the NMS?

Are there hyperparameters that I should tune? I added Spatial2DDropout layers in the classification model. It slows down the cross entropy loss decrease, but does not improve the result much. I would only tune alpha and gamma if the other hyperparameters do not work, since the original authors have already fine tuned those two parameters.

**Comment by hgaiser** [comment]
Closing this as it is not an issue with the code.

That said, try updating tensorflow to the latest version to see if that makes a difference. I've seen it make quite an impact in the past.

**Comment by mariaculman18** [comment]
Hi @zerovirus123, did the update of TensorFlow to the latest version worked for you?
